### PR TITLE
test(js): add test script for jsdom

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

add jsdom test script to `test:coverage` in `js`. not using GitHub Actions matrix since it's for one package only.

## Testing

UT

## Reviewers
@logto-io/eng

### CODE REVIEW

File: packages/js/src/core/fetch-token.test.ts [hunk 1]
```diff
@@ -32,7 +30,7 @@ describe('fetch access token by providing valid refresh token', () => {
         refreshToken: 'refresh_token',
         scope: ['read', 'register', 'manage'],
       },
-      conditional(isNode() && fetchFunction)
+      fetchFunction // Always passing `fetchFunction` since Jest has no `fetch()`
```
Comment (demonzoo) [inline packages/js/src/core/fetch-token.test.ts:33]: But if we run it in jsdom env, there is a fetch implementation by browser, isn't it?

Comment (gao-sun) [reply packages/js/src/core/fetch-token.test.ts:33]: no, jest is running in node so there's no `fetch` until we use a headless browser environment.

Comment (demonzoo) [reply packages/js/src/core/fetch-token.test.ts:33]: If you specify --env=jsdom, then there IS a fetch implementation, right?

Comment (demonzoo) [reply packages/js/src/core/fetch-token.test.ts:33]: OK, just googled it and I'm definetely wrong. Sorry about that 😂

**Review by demonzoo — APPROVED** [review]

**Pull request merged**